pages/arena.php

```diff
@@ -191,6 +195,44 @@ function processArenaBattleEnd(BattleManager|BattleManagerV2 $battle, User $play
             $stat_gain_display .= $player->addStatGain($stat_to_gain, 1) . '.';
         }
 
+        // Village reputation gain adjustments
+        // TODO: Daily cap these
+        if($opponent->level > $player->level) {
+            $lvl_diff = $opponent->level - $player->level;
+
+            // 100%
+            if($level_difference > 7) {
+                $village_rep_chance = 100;
+                if(mt_rand(1, 100) >= 75) {
+                    $village_rep_chance++;
+                }
+            }
+            elseif($level_difference >= 5) {
+                $village_rep_chance += mt_rand(15, 25);
+            }
+            else {
+                $village_rep_chance += mt_rand(5, 10);
+            }
+        }
```

Comment:
Since we're starting reputation from scratch, what if we increased the amount of reputation both granted and required all around so we can give consistent amounts instead of random chance? 

As example for arena this could be like

- 2 or more levels lower = no reputation
- 1 level lower <> 2 levels higher = 1 reputation
- 3+ levels higher = 2 reputation

We'd also need to increase the proportional gains from other sources like PvP and whatnot, as well as the weekly limit and amount required for each reputation tier.

Additionally, we could move this logic to the village class, e.g. a method like`arenaRepGain($player_level, $opponent_level)`